sdk/core/azure-core/src/main/java/com/azure/core/implementation/ResponseConstructorsCache.java

```diff
@@ -0,0 +1,213 @@
+            .findFirst()
+            .flatMap(constructor -> {
+                try {
+                    MethodHandles.Lookup lookup = MethodHandles.lookup();
+                    MethodHandle ctrMethodHandle = lookup.unreflectConstructor(constructor);
+                    switch (constructor.getParameterCount()) {
+                        case 3:
+                            Object f3 = LambdaMetafactory.metafactory(lookup,
+                                    "apply",
+                                    MethodType.methodType(ResponseFunc3.class),
+                                    MethodType.methodType(Object.class,
+                                            Object.class,
+                                            int.class,
+                                            Object.class),
+                                    ctrMethodHandle,
+                                    ctrMethodHandle.type()).getTarget().invoke();
+                            return Optional.of(new ResponseConstructor(3, f3));
+                        case 4:
+                            Object f4 = LambdaMetafactory.metafactory(lookup,
+                                    "apply",
+                                    MethodType.methodType(ResponseFunc4.class),
+                                    MethodType.methodType(Object.class,
+                                            Object.class,
+                                            int.class,
+                                            Object.class,
+                                            Object.class),
+                                    ctrMethodHandle,
+                                    ctrMethodHandle.type()).getTarget().invoke();
+                            return Optional.of(new ResponseConstructor(4, f4));
+                        case 5:
+                            Object f5 = LambdaMetafactory.metafactory(lookup,
+                                    "apply",
+                                    MethodType.methodType(ResponseFunc5.class),
+                                    MethodType.methodType(Object.class,
+                                            Object.class,
+                                            int.class,
+                                            Object.class,
+                                            Object.class,
+                                            Object.class),
+                                    ctrMethodHandle,
+                                    ctrMethodHandle.type())
+                                    .getTarget().invoke();
+                            return Optional.of(new ResponseConstructor(5, f5));
+                        default:
+                            return Optional.<ResponseConstructor>empty();
+                    }
+
+                } catch (Throwable t) {
+                    throw logger.logExceptionAsError(new RuntimeException(t));
+                }
+            }));
+        return this.cache.get(responseClass);
```

Comment:
You don't need to call get - you can just return directly from the `computeIfAbsent` call

Reply:
nice catch!